core/services/fluxmonitor/flux_monitor.go

```diff
@@ -833,44 +786,72 @@ func (p *PollingDeviationChecker) roundState() (contracts.FluxAggregatorRoundSta
 	// It's pointless to listen to logs from before the current reporting round
 	p.reportableRoundID = big.NewInt(int64(roundState.ReportableRoundID))
 
-	// Update the roundTimer using the .TimesOutAt field describing the current round
+	// Update our tickers to reflect the current on-chain round
+	p.resetRoundTimeoutTicker(roundState)
+	p.resetIdleTicker(roundState.StartedAt)
+
+	return roundState, nil
+}
+
+func (p *PollingDeviationChecker) resetRoundTimeoutTicker(roundState contracts.FluxAggregatorRoundState) {
+	loggerFields := []interface{}{
+		"timesOutAt", roundState.TimesOutAt(),
+		"pollFrequency", p.initr.PollTimer.Frequency,
+		"idleDuration", p.initr.IdleTimer.Duration,
+		"mostRecentSubmittedRoundID", p.mostRecentSubmittedRoundID,
+		"reportableRoundID", p.reportableRoundID,
+		"contract", p.initr.Address.Hex(),
+		"jobID", p.initr.JobSpecID.String(),
+	}
+
 	if roundState.TimesOutAt() == 0 {
-		logger.Debugw("updating roundState.TimesOutAt",
-			"value", roundState.TimesOutAt(),
-			"pollDelay", p.pollTicker.d,
-			"idleThreshold", p.idleThreshold,
-			"mostRecentSubmittedRoundID", p.mostRecentSubmittedRoundID,
-			"reportableRoundID", p.reportableRoundID,
-			"contract", p.initr.InitiatorParams.Address.Hex(),
-		)
 		p.roundTimer = nil
+		logger.Debugw("disabling roundTimer, no active round", loggerFields...)
+
 	} else {
-		timeUntilTimeout := time.Unix(int64(roundState.TimesOutAt()), 0).Sub(time.Now())
-		if timeUntilTimeout.Seconds() <= 0 {
+		timesOutAt := time.Unix(int64(roundState.TimesOutAt()), 0)
+		timeUntilTimeout := time.Until(timesOutAt)
+
+		if timeUntilTimeout <= 0 {
 			p.roundTimer = nil
-			logger.Debugw("NOT updating roundState.TimesOutAt, negative duration",
-				"value", roundState.TimesOutAt(),
-				"pollDelay", p.pollTicker.d,
-				"idleThreshold", p.idleThreshold,
-				"mostRecentSubmittedRoundID", p.mostRecentSubmittedRoundID,
-				"reportableRoundID", p.reportableRoundID,
-				"contract", p.initr.InitiatorParams.Address.Hex(),
-			)
+			logger.Debugw("disabling roundTimer, negative duration", loggerFields...)
```

Comment:
"roundTimer has run down; disabling" would be clearer. "negative duration" is kind of an implementation detail.